docs/tutorials/semantic_search/asymmetric_embedding_model.md

```diff
@@ -0,0 +1,539 @@
+---
+
+## Step 1: Spin Up a Docker OpenSearch Cluster
+
+To run OpenSearch in a local development environment, you can use Docker and a pre-configured `docker-compose` file.
+
+### a. Update Cluster Settings
+
+Before proceeding, ensure your cluster is configured to allow registering models. You can do this by updating the cluster settings via the following request:
+
+```
+PUT _cluster/settings
+{
+  "persistent": {
+    "plugins.ml_commons.allow_registering_model_via_url": "true",
+    "plugins.ml_commons.only_run_on_ml_node": "false",
+    "plugins.ml_commons.model_access_control_enabled": "true",
+    "plugins.ml_commons.native_memory_threshold": "99"
+  }
+}
+```
+
+This configuration ensures that OpenSearch can accept machine learning models from external URLs and can run models across non-ML nodes.
+
+### b. Use a Docker Compose File
+
+You can use this sample [file](https://opensearch.org/docs/latest/install-and-configure/install-opensearch/docker/#sample-docker-compose-file-for-development) as an example.
+Once your `docker-compose.yml` file is ready, run the following command to start OpenSearch in the background:
+
+```
+docker-compose up -d
+```
+
+---
+
+## Step 2: Prepare the Model for OpenSearch
+
+In this tutorial, we’ll use the Hugging Face model `intfloat/multilingual-e5-small`, which is capable of generating multilingual embeddings. Follow these steps to prepare and zip the model for use in OpenSearch.
+
+### a. Clone the Model from Hugging Face
+
+To download the model, use the following steps:
+
+1. Install Git Large File Storage (LFS) if you haven’t already:
+
+   ```
+   git lfs install
+   ```
+
+2. Clone the model repository:
+
+   ```
+   git clone https://huggingface.co/intfloat/multilingual-e5-small
+   ```
+
+This will download the model files into a directory on your local machine.
+
+### b. Zip the Model Files
+
+In order to upload the model to OpenSearch, you must zip the necessary model files (`model.onnx`, `sentencepiece.bpe.model`, and `tokenizer.json`). The `model.onnx` file is located in the `onnx` directory of the cloned repository.
+
+Run the following command in the directory containing these files:
+
+```
+zip -r intfloat-multilingual-e5-small-onnx.zip model.onnx tokenizer.json sentencepiece.bpe.model
+```
+
+This command will create a zip file named `intfloat-multilingual-e5-small-onnx.zip`, with the previous mentioned files.
+
+### c. Calculate the Model File Hash
+
+Before registering the model, you need to calculate the SHA-256 hash of the zip file. Run this command to generate the hash:
+
+```
+shasum -a 256 intfloat-multilingual-e5-small-onnx.zip
+```
+
+Make a note of the hash value, as you will need it during the model registration process.
+
+### d. Serve the Model File Using a Python HTTP Server
+
+To allow OpenSearch to access the model file, you need to serve it via HTTP. Since this is a local development environment, you can use Python's built-in HTTP server:
```

Comment:
```suggestion
To allow OpenSearch to access the model file, you need to serve it through HTTP. Because this tutorial uses a local development environment, you can use Python's built-in HTTP server:
```